Classes/KIFUITestActor.m

```diff
@@ -1182,13 +1189,24 @@ - (void)scrollAccessibilityElement:(UIAccessibilityElement *)element inView:(UIV
 - (void)waitForFirstResponderWithAccessibilityLabel:(NSString *)label
 {
     [self runBlock:^KIFTestStepResult(NSError **error) {
-        UIResponder *firstResponder = [[[UIApplication sharedApplication] keyWindow] firstResponder];
-        if ([firstResponder isKindOfClass:NSClassFromString(@"UISearchBarTextField")]) {
-            do {
-                firstResponder = [(UIView *)firstResponder superview];
-            } while (firstResponder && ![firstResponder isKindOfClass:[UISearchBar class]]);
+        BOOL didMatch = NO;
+        UIResponder *foundResponder = nil;
+        for (UIResponder  *firstResponder in [[UIApplication sharedApplication] firstResponders]) {
+            foundResponder = firstResponder;
+
+            if ([foundResponder isKindOfClass:NSClassFromString(@"UISearchBarTextField")]) {
+                do {
+                    foundResponder = [(UIView *)foundResponder superview];
+                } while (foundResponder && ![foundResponder isKindOfClass:[UISearchBar class]]);
+            }
+
+            if (foundResponder.accessibilityLabel == label || [foundResponder.accessibilityLabel isEqualToString:label]) {
+                didMatch = YES;
+                break;
+            }
         }
-        KIFTestWaitCondition([[firstResponder accessibilityLabel] isEqualToString:label], error, @"Expected accessibility label for first responder to be '%@', got '%@'", label, [firstResponder accessibilityLabel]);
+
+        KIFTestWaitCondition(didMatch, error, @"Expected accessibility label for first responder to be '%@', got '%@'", label, [foundResponder accessibilityLabel]);
```

Comment:
Maybe the assertions should use dump the whole array of elements (or maybe at least the count of elements checked)? (same applies below)